### DISCUSSION TITLE
rag for json file

### DISCUSSION THREAD

**Discussion by codeMonkey-shin** [open-webui/open-webui#3750] [General]
Could you create a new feature to handle JSON files in the Rag system?

**Comment by atgehrhardt** [comment]
Not sure what you mean here, can you share more information? Currently the RAG system doesn't support the json extension, BUT, will accept the file as text. 

You'll receive a notification when uploading: 
"Unknown File Type 'application/json', but accepting and treating as plain text"

It WILL upload the file however, and you can interact with it via rag. Because almost every LLM out there understands json quite well, there's no shortcomings to this approach that I am aware of.
